### DISCUSSION TITLE
Installation Fail

### DISCUSSION THREAD

**Discussion by jkaidor** [magicbug/Cloudlog#3005] [General]
Hello!  
   I'm excited about Cloudlog - it looks like exactly what I've been looking for.  Unfortunately, it doesn't like my LAMP server.    It's a Slackware 14 machine, and the PHP version is 5.6.23.    The install script runs without issue, and the "cloudlog" database has been populated.  But the dashboard fails:
----------------- snip ---------------------
A PHP Error was encountered
Severity: Parsing Error
Message: syntax error, unexpected '>'
Filename: controllers/Dashboard.php
Line Number: 163
--------------- endsnip ------------------------
I'm guessing that it's choking on the "spaceship" operator in line 163.  Not supported in that old PHP.

This is a production system;  my business depends on it, and I am VERY reluctant to try to upgrade its PHP.

I have another machine with Slackware 15 that I am gradually bringing on line.  It has PHP 7.4.27.  But it does not have a public IP yet.  It's behind the main machine on the localnet.

I wonder if there's a way to have the main machine sluff off certain http requests to the newer localnet machine?

I actually wrote a logging application for my personal use.  Primitive.  I just added it to my business tools, where I had infrastructure.  Wrote it in Perl.   I know almost nothing about PHP, but I've written something like
200K lines of Perl in the past couple decades.  I really don't want to expand my own program.  Cabrillo, LOTW etc...  I'd like to do some actual hamming while the sunspots are good :).

  I installed Cloudlog on the localnet machine.  It gets past the spaceship operator.  But it complains that the first table in cloudlog database already exists.  OK,  I guess it's trying to run the install script.  So I deleted the database.   Now it complains that the database doesn't exist.  

Guess I should read the documentation :).

           - Jerry Kaidor, KF6VB

**Comment by phl0** [comment]
The spaceship operator was introduced in PHP7 which is the minimum required version here.
You could try using this code snippet as a workaround:

```
if ($dateA == $dateB) {
   return 0;
} else if ($dateA < $dateB) {
   return -1;
} else if ($dateA > $dateB) {
   return 1;
}
```

See https://www.php.net/manual/en/migration70.new-features.php.

**Comment by bgritzmacher** [comment]
> I wonder if there's a way to have the main machine sluff off certain http requests to the newer localnet machine?

I suppose, yes, by utilizing php-fpm on a different machine and proxy requests to it would achieve that, and nothing seems to suggest that you *cannot* do that (for example, by setting up an nginx + php environment in docker, you do basically that.) But doing it that way with bare metal or separate virtual machines is more complicated and has its fair share of caveats that you really need to investigate first.

I bring this up more as food for thought than as an actual serious suggestion. For all I know, someone may actually have this kind of setup and can bring in some insight!

**Reply by jkaidor** [reply]
OK - I deleted some stuff ( forget exactly what ) and tried again, on my localnet machine - the one with php 7.something.
This time, it died at line 20 of 082_add_reset_pass_to_users.php.  It was trying to add a reset_password_code field
to the "users" table, and died because that field already exists.
Instead of digging back and figuring out exactly how it got there - or maybe deleting EVERYTHING & starting over again, I just commented out that line, hit refresh on the browser, et voila!  A login screen!
